python/orion_finance_sdk_py/costs/venues/uniswap_v3/simulator.py

```
@@ -0,0 +1,258 @@
+        amount_out_human = size_abs
+    else:
+        # Exact-input of the asset: asset in, USDC out.
+        zero_for_one = asset_is_token0
+        amount_in_total, amount_out_raw, fee_raw, _ = _walk_swap(
+            state, amount_asset_raw, zero_for_one
+        )
+        if amount_in_total < amount_asset_raw:
+            raise ValueError(
+                f"Pool {state.meta.address} cannot fill exact output "
+                f"{amount_asset_raw} of {asset}; filled {amount_in_total}"
+            )
```

Ruff (0.16.1)

[warning] Avoid specifying long messages outside the exception class

(TRY003)